server/src/main/java/org/apache/druid/segment/realtime/appenderator/DefaultOfflineAppenderatorFactory.java

```diff
@@ -36,24 +37,27 @@
   private final ObjectMapper objectMapper;
   private final IndexIO indexIO;
   private final IndexMerger indexMerger;
+  private final boolean useOak;
 
   @JsonCreator
   public DefaultOfflineAppenderatorFactory(
       @JacksonInject DataSegmentPusher dataSegmentPusher,
       @JacksonInject ObjectMapper objectMapper,
       @JacksonInject IndexIO indexIO,
-      @JacksonInject IndexMerger indexMerger
+      @JacksonInject IndexMerger indexMerger,
+      @JacksonInject @Named("useOak") boolean useOak
```

Comment:
what is the meaning of using '@Named'? that it is externally configurable?

Reply:
It's for mapping bindings without custom annotations. I use it in the [Peon module configuration](https://github.com/yonigottesman/incubator-druid/pull/2/files/aecf215d33fa4202d7889fbb6b16b5d00eeed0e7..b3389ff68a6203331c32f17d59e087ee8ef0e889#diff-1f5104ab05f9227d0c91438392b45cbeR203):
```
binder.bindConstant().annotatedWith(Names.named("useOak")).to(Boolean.parseBoolean(properties.getProperty("druid.indexer.useOak")));
```


I'm not a guice expert, this just seemed like the easiest way to bind a primitive value without wrapping it up in a class or a custom factory.